### DISCUSSION TITLE
ECharts-Stat

### DISCUSSION THREAD

**Discussion by humphreylee** [yuankunzhang/charming#127] [Q&A] [answered]
Is it possible to use [ECharts-Stat](https://github.com/ecomfe/echarts-stat) or any [other](https://github.com/ecomfe/echarts-gl) ECharts extension with Charming? If so, could you share an example?

**Comment by LukaOber** [comment, marked as answer]
Echarts-Stat is not implemented. I think minimal use of echarts-gl is possible. There is a bar3D example which, I believe, works using echarts-gl. 

The features for 3D Plots are not nearly complete. For complex plots you would probably need to implement some functionalities yourself.

**Comment by humphreylee** [comment]
Curious about the effort needed to make the ECharts-Stat extension functional in Charming, if it’s feasible?

**Comment by LukaOber** [comment]
The heavy lifting is done by echarts. I don't think this would be very complicated, as you are mostly just setting some options. Would you be interested in implementing a first version?

**Comment by humphreylee** [comment]
I’ll give it a shot. I’m looking to create a scatter plot with marginal histograms similar to the example [here](https://github.com/ecomfe/echarts-stat). Any tips would be appreciated.
